internal/kafka/internal/services/kafka.go

```diff
@@ -960,52 +962,107 @@ type KafkaStatusCount struct {
 	Count  int
 }
 
-type KafkaRegionCount struct {
+type KafkaStreamingUnitCountPerRegion struct {
 	Region        string
-	InstanceType  string `gorm:"column:instance_type"`
-	ClusterId     string `gorm:"column:cluster_id"`
+	InstanceType  string
+	ClusterId     string
 	Count         float64
-	CloudProvider string `gorm:"column:cloud_provider"`
+	CloudProvider string
 }
 
-func (k *kafkaService) CountByRegionAndInstanceType() ([]KafkaRegionCount, error) {
+func (kafkaStreamingUnitCountPerRegion KafkaStreamingUnitCountPerRegion) isSame(kafkaPerRegionFromDB *KafkaPerRegionCount) bool {
+	return kafkaStreamingUnitCountPerRegion.CloudProvider == kafkaPerRegionFromDB.CloudProvider &&
+		kafkaStreamingUnitCountPerRegion.ClusterId == kafkaPerRegionFromDB.ClusterId &&
+		kafkaStreamingUnitCountPerRegion.InstanceType == kafkaPerRegionFromDB.InstanceType &&
+		kafkaStreamingUnitCountPerRegion.Region == kafkaPerRegionFromDB.Region
```

Comment:
Do we need to check cloud provider and region? would `ClusterId` + `InstanceType` be sufficient? `ClusterId` should be unique and `InstanceType` can be checked if the cluster supports multiple instances types :thinking:

Reply:
I think cluster Id and instance type should be enough.
I kept the same condition as the one which was there previously